### PR TITLE
feat(controller): add rpc methods for controller

### PR DESCRIPTION
**Pull request by heiruwu**
Because

- support controller service for model resource state monitoring

This commit

- add rpc methods for controller
- update admin/private wording
- fix trailing whitespaces and end of file line

### CODE REVIEW

File: vdp/model/v1alpha/model_private_service.proto [hunk 1]
```diff
@@ -12,7 +12,7 @@ import "vdp/model/v1alpha/model.proto";
 service ModelPrivateService {
   option (google.api.default_host) = "api.instill.tech";
 
-    // ========== Admin API ========== 
+    // ========== Admin API ==========
```
Comment (pinglin) [inline vdp/model/v1alpha/model_private_service.proto:15]: We probably wanna name these APIs `Private API`. Need to change this in other backend protobuf comment too. (cc'd @xiaofei-du)

**Review by pinglin — CHANGES_REQUESTED** [review]

File: vdp/controller/v1alpha/controller.proto [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    enum State {
+      // State: UNSPECIFIED
+      STATE_UNSPECIFIED = 0;
+      // State: OFFLINE
+      STATE_OFFLINE = 1;
+      // State: ONLINE
+      STATE_ONLINE = 2;
+      // State: ERROR
+      STATE_ERROR = 3;
+    }
```
[the remark this answers is not in the document]
Comment (xiaofei-du) [reply vdp/controller/v1alpha/controller.proto:20]: agree

File: vdp/model/v1alpha/model_public_service.proto [hunk 3]
```diff
@@ -208,6 +208,16 @@ service ModelPublicService {
     option (google.api.method_signature) = "name";
   }
 
+  // WatchModelInstance method receives a WatchModelInstanceRequest message
+  // and returns a WatchModelInstanceResponse
+  rpc WatchModelInstance(WatchModelInstanceRequest)
+      returns (WatchModelInstanceResponse) {
+    option (google.api.http) = {
+      get : "/v1alpha/{name=models/*/instances/*/watch}"
```
Comment (xiaofei-du) [inline vdp/model/v1alpha/model_public_service.proto:216]:
```suggestion
      get : "/v1alpha/{name=models/*/instances/*}/watch"
```

File: vdp/controller/v1alpha/controller_service.proto [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  rpc GetResource(GetResourceRequest) returns (GetResourceResponse);
+
+  // UpdateResource method receives a UpdateResourceRequest message
+  // and returns a UpdateResourceResponse
+  rpc UpdateResource(UpdateResourceRequest) returns (UpdateResourceResponse);
+
+  // DeleteResource method receives a DeleteResourceRequest message
+  // and returns a DeleteResourceResponse
+  rpc DeleteResource(DeleteResourceRequest) returns (DeleteResourceResponse);
+}
```
Comment (xiaofei-du) [inline vdp/controller/v1alpha/controller_service.proto:31]: Can we follow how `pipeline/v1alpha/pipeline_public_service.proto` defines the service using the Google API annotations like `google.api.http` and `google.api.method_signature` to pin down the endpoint details?

**Review by xiaofei-du — CHANGES_REQUESTED** [review]

File: vdp/controller/v1alpha/controller.proto (hunk 2 above)
[the remark this answers is not in the document]
Comment (heiruwu) [reply vdp/controller/v1alpha/controller.proto:20]: What I am thinking is, we refactored the `enum State` in each resources' proto to have only `STATE_ONLINE ` and `STATE_OFFLINE`, and only the `enum State` in controller.proto has these four states. `oneof` will definitely works too

[the remark this answers is not in the document]
Comment (pinglin) [reply vdp/controller/v1alpha/controller.proto:20]: That can certainly do but just not semantically straightforward for pipeline and connector to be `online/offline` compared with [`active/inactive`](https://www.instill.tech/docs/core-concepts/pipeline#state) and [`connected/disconnected`](https://www.instill.tech/docs/core-concepts/connector#state) respectively.

[the remark this answers is not in the document]
Comment (heiruwu) [reply vdp/controller/v1alpha/controller.proto:20]: Agree

**Review by pinglin — APPROVED** [review]
LGTM

**Pull request merged**